pkg/config/sampling/sampling.go

```diff
@@ -81,6 +82,14 @@ func (u *Config) Get() *corev1.ConfigMap {
 // support for the Sampling configmap.
 func Update(jaeger *v1.Jaeger, commonSpec *v1.JaegerCommonSpec, options *[]string) {
 
+	// Check to validate if sampling strategy file is already passed as an option
```

Comment:
I think the `Get()` function above should also return `nil` (ConfigMap) if the option is explicitly defined, so that an unused config map is not created which might be misleading. So would be better if this check is in a separate function that could also be used in the `Get()` - which would then need to check the options in the `jaeger` arg (from either allInOne or collector node, depending upon the `strategy`).

Reply:
But, isn't [this statement](https://github.com/jaegertracing/jaeger-operator/blob/fd8a3fa038282c619cc796e7bc4ec9f7e01c10ce/pkg/config/sampling/sampling.go#L33) checking for the sampling options in the Get function? Based on my understanding the `jaeger.Spec.Sampling.Options` is same as the options which we are checking in the update function. Please correct me if my understanding is incorrect.

Reply:
That statement is just checking what content should be placed in the ConfigMap. So it would be possible for someone to specify the sampling config in the Jaeger CR, and still provide an option specifying their own `sampling.strategies-file` parameter - in which case I believe we are allowing the parameter to take precedence.